src/shellhistory/shellhistory.sh

```diff
@@ -125,23 +139,27 @@ _shellhistory_after() {
 }
 
 _shellhistory_enable() {
-  # mkdir -p "${SHELLHISTORY_ROOT}" &>/dev/null
+  _SHELLHISTORY_BEFORE_DONE=2
+  _SHELLHISTORY_AFTER_DONE=1
   if [ "${ZSH_VERSION}" ]; then
-    _shellhistory_command_type() { _shellhistory_zsh_command_type "$1"; }
       # FIXME: don't override possible previous contents of precmd
+    preexec() { _shellhistory_before; }
     precmd() { _shellhistory_after; }
   elif [ "${BASH_VERSION}" ]; then
-    _shellhistory_command_type() { _shellhistory_bash_command_type "$1"; }
-    PROMPT_COMMAND='_shellhistory_after;'$'\n'"${PROMPT_COMMAND}"
+    PROMPT_COMMAND="_shellhistory_after; ${PROMPT_COMMAND}"
+    trap '_shellhistory_before' DEBUG
```

Comment:
Is there a way to retain previous contents of the trap and append new contents without overriding?

Reply:
Done in https://github.com/pawamoy/shell-history/pull/23/commits/7336a8c347fb0d593b74e3b501f51cf484eb4afd